src/client/Microsoft.Identity.Client/Instance/AuthorityManager.cs

```diff
@@ -46,25 +54,29 @@ public async Task RunInstanceDiscoveryAndValidationAsync()
                                     metadata.PreferredNetwork);
 
                 // We can only validate the initial environment, not regional environments
-                await _requestContext.ServiceBundle.AuthorityEndpointResolutionManager.ValidateAuthorityAsync(
-                    _initialAuthority, 
-                    _requestContext).ConfigureAwait(false);
+                await ValidateAuthorityAsync(_initialAuthority).ConfigureAwait(false);
 
                 _instanceDiscoveryAndValidationExecuted = true;
             }
         }
 
-        public AuthorityEndpoints GetEndpoints(string loginHint)
+        public static /* for test */ void ClearValidationCache()
         {
-            if (!_instanceDiscoveryAndValidationExecuted)
+            s_validatedEnvironments.Clear();
+        }
+      
+        private async Task ValidateAuthorityAsync(Authority authority)
+        {
+            // race conditions could occur here, where multiple requests validate the authority at the same time
+            // but this is acceptable and once the cache is filled, no more HTTP requests will be made
+            if (!s_validatedEnvironments.Contains(authority.AuthorityInfo.Host))
```

Comment:
why would that be acceptable in this case?

Reply:
We have 2 options: 

1. (status quo) - no syncronization. Multiple validation requests will be made (when system is starting), until the cache fills up (~1 second worth of requests, so we're talking few requests). 
2. use syncronization, i.e. make all requests wait for the one doing the instance dicsovery. As we found out with MyApps, if that request times out or takes ~30s to complete, then all requests will be blocked for that interval and the system is paralyzed.

To me, option 1 seems less bad.